docs/pages/reference/terraform/resources/saml_connector.mdx

```diff
@@ -0,0 +1,146 @@
+---
+title: "teleport_saml_connector Resource - teleport"
+description: |-
+  
+---
+
+# teleport_saml_connector (Resource)
+
+
+
+## Example Usage
+
+```hcl
+# Teleport SAML connector
+# 
+# Please note that SAML connector will work in Enterprise version only. Check the setup docs:
+# https://goteleport.com/docs/enterprise/sso/okta/
+
+resource "teleport_saml_connector" "example" {
+  # This block will tell Terraform to never update private key from our side if a keys are managed 
+  # from an outside of Terraform.
+
+  # lifecycle {
+  #   ignore_changes = [
+  #     spec[0].signing_key_pair[0].cert,
+  #     spec[0].signing_key_pair[0].private_key,
+  #     spec[0].assertion_key_pair[0].cert,
+  #     spec[0].assertion_key_pair[0].private_key,
+  #   ]
+  # }
+
+  # This section tells Terraform that role example must be created before the SAML connector
+  depends_on = [
+    teleport_role.example
+  ]
+
+  metadata = {
+    name = "example"
+  }
+
+  spec = {
+    attributes_to_roles = [{
+      name  = "groups"
+      roles = ["example"]
+      value = "okta-admin"
+      },
+      {
+        name  = "groups"
+        roles = ["example"]
+        value = "okta-dev"
+    }]
+
+    acs               = "https://localhost:3025/v1/webapi/saml/acs"
+    entity_descriptor = ""
+  }
+}
+```
+
+{/*  schema generated by tfplugindocs */}
+## Schema
+
+### Required
+
+- `spec` (Attributes) Spec is an SAML connector specification. (see [below for nested schema](#nested-schema-for-spec))
+- `version` (String) Version is the resource version. It must be specified. Supported values are: `v2`.
+
+### Optional
+
+- `kind` (String) Kind is a resource kind.
+- `metadata` (Attributes) Metadata holds resource metadata. (see [below for nested schema](#nested-schema-for-metadata))
+- `sub_kind` (String) SubKind is an optional resource sub kind, used in some resources.
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+### Nested Schema for `spec`
+
+Required:
+
+- `acs` (String) AssertionConsumerService is a URL for assertion consumer service on the service provider (Teleport's side).
```

Comment:
There are some instances (like this one) where the Go identifier is meaningfully different than the Terraform provider field name. Is there a way to replace Go identifier names with the Terraform field name, e.g., substituting `acs` for `AssertionConsumerService`?

Reply:
I don't think there's an easy way to do it. We pick up the comments from the proto messages and building the substition map will require complex changes in protoc-gen-terraform. I'll see what I can do but I don't have much hope